### PR TITLE
[Llama2] Fix setting seeds of distributed training for Llama model with Auto Parallel

### PR DESCRIPTION
**Pull request by haohongxiang**
### PR types

Bug fixes
### PR changes

Others
### Description

[Llama2] Fix setting seeds of distributed training for Llama model with Auto Parallel

### CODE REVIEW

File: paddlenlp/trainer/trainer_utils.py [hunk 1]
```diff
@@ -57,14 +59,37 @@
 ]
 
 
-def set_seed(seed: int = 1234):
+def get_dist_seeds(seed: int = 1234, topo: Topology = None):
```
Comment (ZHUI) [inline paddlenlp/trainer/trainer_utils.py]:
```suggestion
def _get_distributed_seeds(seed: int = 1234, topo: Topology = None):
```

File: paddlenlp/trainer/trainer_utils.py [hunk 2]
```diff
@@ -57,14 +59,37 @@
...
+    """
+    Get the seeds from distributed environment strategy.
+    Args:
+        seed (:obj:`int`, `optional`, defaults to 1234): The seeds for initializing distributed training.
+        topo (:obj:`Topology`, `optional`, defaults to None): The topology of hybrid parallel in semi-auto mode.
+    Returns:
+        Tuple[int, int]: The global seed and local seed respectively.
+    """
```
Comment (ZHUI) [inline paddlenlp/trainer/trainer_utils.py:70]: 这个函数里面不仅有get_dist_seeds，还有random和numpy的seed设置。
```python
    random.seed(seed + 100 * pp_rank)
    np.random.seed(seed + 100 * pp_rank)
```

File: paddlenlp/trainer/trainer_utils.py (hunk 1 above)
Comment (haohongxiang) [reply paddlenlp/trainer/trainer_utils.py]: Done

File: paddlenlp/trainer/trainer_utils.py (hunk 2 above)
Comment (haohongxiang) [reply paddlenlp/trainer/trainer_utils.py:70]: 已抽到`function set_seed( )`中

**Review by ZHUI — APPROVED** [review]
LGTM

**Review by zhaoyinglia — APPROVED** [review]
LGTM

**Pull request merged**